### PR TITLE
[feat] uplift tests for ~95% coverage

### PR DESCRIPTION
**Pull request by ben-ranford**
## Issue Being Solved
The project had limited branch and error-path test coverage across analyzers, adapters, and UI flows, making regressions easier to ship.

## Cause and Effect on Users
When edge cases are under-tested, users can see broken parsing, incorrect recommendations, or unstable report output after routine refactors. That raises upgrade risk and reduces trust in CI signals.

## Root Cause
Coverage depth was concentrated on happy paths, and the coverage gate was not strict enough to force broad test additions as the codebase evolved.

## Fix Implemented
This branch now focuses on test and quality uplift only:
- Adds broad branch/error-path unit coverage across analysis, CLI parsing, JS/JVM/Python adapters, language/report/runtime/safeio/workspace, and UI command paths.
- Raises the enforced coverage gate to 90% (Makefile + CI alignment) so the expanded baseline is protected going forward.
- Removes the previously included tunable-threshold/config-override feature work from this branch.

## Validation
- `make ci`
- Pre-commit suite (`make fmt`, `make ci`, `make cov`)
- Coverage result from `make cov`: **91.3% total** with required threshold **>= 90%**

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Intentionally left without CPD exclusions so test files are included in
+# duplication analysis.
```
[the remark this answers is not in the document]
Comment (ben-ranford) [reply sonar-project.properties]: oops didn't mean to commit that
